### PR TITLE
Do not initialize the prismic API in static mode.

### LINKED ISSUE #111: Support the new target: "static"
The target "static" has been introduced in nuxt 1.13. It would be nice if prismic-module would support it.

Right now, using the static target, there is one call to the prismic api in the generated site:
https://github.com/nuxt-community/prismic-module/blob/c824747c3e90634324236d897097da11e7a2d831/src/templates/plugins/prismic.js#L19

### PR DESCRIPTION
**Pull request by francois2metz**
This avoid an useless call to the Prismic API.

Fix #111

### CODE REVIEW

File: src/templates/plugins/prismic.js [hunk 1]
```diff
@@ -65,7 +71,7 @@ export default async (context, inject) => {
           return PrismicDOM.Date(date)
         }
       },
-      <% if (options.preview) { %>
+      <% if (!(process.client && process.static) && options.preview) { %>
```
Comment (VincentHardouin) [inline src/templates/plugins/prismic.js:74]:
```suggestion
      <% if (!isClientStatic && options.preview) { %>
```

Comment (lihbr) [reply src/templates/plugins/prismic.js:74]: Nice catch!

Comment (lihbr) [reply src/templates/plugins/prismic.js:74]: Doesn't this kill the preview behavior once in production though?

**Comment by lihbr** [comment]
Hey, thank you so much for this PR! I hope we'll manage to land something like that on production as removing this API call seemed a bit more complicated than that for me hmm 🤔

1. One thing I'm concerned about is what if you have some pages that aren't static? (i.e. relying on an SPA behavior because you don't care about SEO on those) In this case, even with a static behavior, you'll want the API to be available. The same goes if you fetch some content outside of `asyncData` or `fetch`, like on a method or a lifecycle, or when you want to use the preview.

2. I had this API issue at the back of my mind for a while now and I was thinking a way to fix it would be more about initing it only when first-called rather than looking at the current runtime/environment. I'm not sure although to what degree `process.*` constants change once in production:
    - is `process.static` still `true` when in SPA fallback;
    - same for `process.client`.

    I'll investigate this part.

3. If breaking changes are needed we can consider rolling a final major of this module for Nuxt 2 (with this upgrade along with others in the backlog) before upgrading it for Nuxt 3 (ca. April 2021) in another major. Although given the tight timeframe I'm leaning toward avoiding, if possible, a major for Nuxt 2 considering the great stability of this version and the extra maintenance it will involve (alongside the one for Nuxt 3).

Cheers~

**Comment by francois2metz** [comment]
>  1. One thing I'm concerned about is what if you have some pages that aren't static? (i.e. relying on an SPA behavior because you don't care about SEO on those) In this case, even with a static behavior, you'll want the API to be available. The same goes if you fetch some content outside of `asyncData` or `fetch`, like on a method or a lifecycle, or when you want to use the preview.

Good point. We may add a new option `fullStatic` or something similar to keep the compatibility and make this possible.

File: src/templates/plugins/prismic.js (hunk 1 above)
Comment (francois2metz) [reply src/templates/plugins/prismic.js:74]: Definitely. See my `fullStatic` proposal below.

**Comment by lihbr** [comment]
I'd be happy with that and perhaps initing again the API if the preview function is triggered to preserve the preview behavior 🙂

**Comment by lihbr** [comment]
Closing in favor of #134

We'll consider switching to the `.client` method soon. I'm really sorry we couldn't lead this pull request to a merge but thank you for your contributions and your interest in Prismic 🙂

**Pull request closed without merging**